File: src/main/scala/committee/nova/vlzoomer/VeryLegacyZoomer.scala
package committee.nova.vlzoomer

import committee.nova.vlzoomer.VeryLegacyZoomer.MODID
import committee.nova.vlzoomer.client.config.ClientConfig
import committee.nova.vlzoomer.client.handler.ClientHandler
import cpw.mods.fml.common.Mod
import cpw.mods.fml.common.Mod.EventHandler
import cpw.mods.fml.common.event.FMLPreInitializationEvent
import cpw.mods.fml.relauncher.Side
import org.apache.logging.log4j.LogManager

@Mod(modid = MODID, useMetadata = true, modLanguage = "scala",
  acceptableRemoteVersions = "*", guiFactory = "committee.nova.vlzoomer.client.gui.factory.GuiFactory")
object VeryLegacyZoomer {
  final val MODID = "vlzoomer"
  final val LOGGER = LogManager.getLogger(MODID)

  @EventHandler def preInit(e: FMLPreInitializationEvent): Unit = {
    if (e.getSide != Side.CLIENT) LOGGER.warn("This mod should only be installed on client.")
    else {
      ClientConfig.init(e.getSuggestedConfigurationFile)
      ClientHandler.init()
    }
  }
}

File: src/main/java/committee/nova/vlzoomer/VeryLegacyZoomer.java
package committee.nova.vlzoomer;

import committee.nova.vlzoomer.client.config.ClientConfig;
import committee.nova.vlzoomer.client.event.ClientHandler;
import cpw.mods.fml.common.Mod;
import cpw.mods.fml.common.event.FMLPreInitializationEvent;
import cpw.mods.fml.common.network.NetworkMod;
import cpw.mods.fml.relauncher.Side;

import java.util.logging.Logger;

@Mod(modid = VeryLegacyZoomer.MODID, useMetadata = true)
@NetworkMod(clientSideRequired = true)
public class VeryLegacyZoomer {
    public static final String MODID = "vlzoomer";

    @Mod.EventHandler
    public void preInit(FMLPreInitializationEvent e) {
        final Logger logger = e.getModLog();
        if (e.getSide() != Side.CLIENT) {
            logger.warning("This mod should only be installed on client.");
            return;
        }
        ClientConfig.init(e.getSuggestedConfigurationFile());
        ClientHandler.init();
    }
}
